ts/src/header-validator/validate-json.ts

```diff
@@ -759,6 +767,32 @@ function channelCapacity(s: Source, ctx: SourceContext): void {
   }
 }
 
+function validateAttributionScopeFields(
+  s: Source,
+  ctx: SourceContext
+): boolean {
+  return ctx.scope('attribution_scope_limit', () => {
+    if (s.attributionScopeLimit === null) {
+      if (s.attributionScopes.size !== 0) {
+        ctx.error('must be set if attribution_scopes is set')
+        return false
+      }
+      if (s.maxEventStates !== constants.defaultMaxEventStates) {
+        ctx.error('must be set if non-default max_event_states value is set')
+        return false
+      }
+      return true
+    }
+    return isInRange(
+      s.attributionScopes.size,
+      ctx,
+      1,
+      s.attributionScopeLimit,
+      `attribution_scope_limit must be greater than or equal to the number of attribution scopes ${s.attributionScopeLimit}`
```

Comment:
```suggestion
      `attribution_scope_limit must be greater than or equal to the number of attribution scopes (${s.attributionScopeLimit})`
```

Reply:
But I'm also confused about the wording here, since it looks like we're validating the size of the `attribution_scopes` field, not the value of the `attribution_scope_limit` field. In other words, I'd expect this to be phrased as something like `number of attribution scopes (<number>) must be less than or equal to attribution_scope_limit (<number>)`.

Reply:
that's my initial thought as well. could you advise how do we change the field scope to attribution_scopes inside another scope? I was having trouble make it work somehow.

Reply:
You won't be able to *change* the scope within a `scope` call, only push a new one, so you'd have to check this outside the outer `ctx.scope` call.

Reply:
thanks, as Nan suggested, I was able to remove this validation method after passing attributionScopeLimit value into `attributionScopes` and `maxEventStates`.